code/modules/antagonists/blob/blob_mobs.dm

```diff
@@ -208,16 +209,45 @@
 			blob_head_overlay.color = overmind.blobstrain.complementary_color
 		color = initial(color)//looks better.
 		add_overlay(blob_head_overlay)
+/mob/living/simple_animal/hostile/blob/blobspore/Goto(target, delay, rally)
+	var/movement_steps = 0
+	if(rally)
+		in_movement = TRUE
 
-/mob/living/simple_animal/hostile/blob/blobspore/Goto(target, delay)
 	if(target == src.target)
 		approaching_target = TRUE
 	else
 		approaching_target = FALSE
-	for(var/w in get_path_to(src, target, simulated_only = FALSE, avoid_mobs = TRUE))
+
+	for(var/w in get_path_to(src, target, simulated_only = FALSE))
+		if(in_movement && !rally) //incase the spore is already chasing something like a player but the rally command is called
+			return
+		movement_steps++
 		step(src, get_dir(src, w))
 		sleep(delay)
 
+	if(!movement_steps) //pathfinding fallback in case we cannot find a valid path at the first attempt
+		var/ln = get_dist(src, target)
+		var/target_new = target
+		var/found_blocker
+		while(!movement_steps && (ln > 0)) //will stop if we can find a valid path or if ln gets reduced to 0 or less
+			for(var/i in 1 to ln) //calling get_path_to every time is quite taxing lets see if we can find whatever blocks us
+				target_new = get_step(target_new,  get_dir(target_new, src)) //step towards the origin until we find the blocker then 1 further
+				ln--
+				if(istype(target_new,/turf/closed) || (locate(/obj/structure/window) in target_new) || (locate(/obj/machinery/door) in target_new) || (locate(/obj/structure/table) in target_new) || (locate(/obj/mecha) in target_new)) //we check for possible things that could block us
```

Comment:
If anyone has a better idea how to do this part i am very much open to suggestions

Reply:
i mean line 237

Reply:
for var/stuff in turf.contents if istype in list

Reply:
> for var/stuff in turf.contents if istype in list

Already tried that by benchmarking it for some reason it seems to be faster to do this with locate but that makes no sense at all honestly